dipy/segment/cythonutils.pyx

```diff
@@ -104,3 +111,46 @@ cdef int same_shape(Shape shape1, Shape shape2) nogil:
         same_shape &= shape1.dims[i] == shape2.dims[i]
 
     return same_shape
+
+
+cdef Data2D* create_memview(Py_ssize_t buffer_size, Py_ssize_t dims[MAX_NDIM]) nogil:
```

Comment:
Do we only want to support 2D memview? If so, we could rename the function so that it is clearer it is only for 2D array.

Reply:
ok, you are right, let's do it more generic

Reply:
Finally, I choose the easy path: renaming. The second option needs more thinking and we need to move forward with the release